src/rgw/cls_fifo_legacy.cc

```diff
@@ -876,8 +877,9 @@ int FIFO::open(lr::IoCtx ioctx, std::string oid, std::unique_ptr<FIFO>* fifo,
   fifo::info info;
   std::uint32_t size;
   std::uint32_t over;
-  int r = get_meta(ioctx, std::move(oid), objv, &info, &size, &over, 0, y);
-  if (r < 0) {
+  int r = get_meta(ioctx, std::move(oid), objv, &info, &size, &over, 0, y,
+		   probe);
+  if (r < 0 && !(probe && (r == -ENOENT || r== -ENODATA))) {
```

Comment:
don't we still want to `return r;` in this error case? `probe` should only affect whether we log an error

Reply:
Yes, thank you.

The power of code review!

Reply:
Fixed.